src/main/java/com/palantir/ri/ResourceIdentifier.java

```diff
@@ -162,7 +145,35 @@ public boolean equals(Object obj) {
      *         {@code false} otherwise.
      */
     public static boolean isValid(String rid) {
-        return rid != null && SPEC_PATTERN.matcher(rid).matches();
+        if (rid == null) {
+            return false;
+        }
+
+        if (!rid.startsWith(RID_PREFIX)) {
+            return false;
+        }
+
+        int serviceIndex = getServiceIndex(rid, RID_PREFIX.length());
```

Comment:
If we extract `RID_PREFIX.length()` into a true constant (`private static final int RID_PREFIX_LENGTH = 3;`, no evaluation on load) it will be inlined directly into uses.